File: vzanHouTai/User.MiniApp/IM/freechat/app/model/immessage.js
var sd = require('silly-datetime');
var time = sd.format(new Date(), 'YYYY-MM-DD HH:mm:ss');
//console.log(time);
/**
 * @description 消息记录类
 */
class immessage {
    constructor(Id = 0, appId = '', aId = 0, storeId = 0, roomId = 0, fuserId = 0, tuserId = 0, msgType = 0, state = 0, msg = '', tuserType=0,ids='',sendDate = time) {
        this.id = Id;//自增id
        this.appId = appId; //小程序appid
        this.aId = aId; //小程序aid
        this.storeId = storeId; //门店id
        this.roomId = roomId; //房间id（暂时未用到）
        this.fuserId = fuserId;//发送者id (群发送，即系统消息为0)
        this.tuserId = tuserId;//接受者id (群接收为0)
        this.msgType = msgType;//消息类型 (-1:系统消息离开，0:文本，1:图像, 2:语音, 3:短视频，4:系统消息进入)
        this.state = state; //状态 默为0 (0:正常  -1:删除)
        this.msg = msg; //消息内容
        this.sendDate = sendDate; //发送时间
        this.tuserType=tuserType;//用户类型
        this.ids=ids;//唯一值 uuid
    }
    setModel(appId = '', aId = 0, storeId = 0, roomId = 0, fuserId = 0){
        this.appId = appId; 
        this.aId = aId; 
        this.storeId = storeId; 
        this.roomId = roomId;  
        this.fuserId = fuserId;
        return this;
    }
}
exports.model = new immessage();

File: vzanHouTai/Entity.MiniApp/Im/ImMessage.cs
using Entity.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Utility;

namespace Entity.MiniApp.Im
{
    [Serializable]
    [SqlTable(dbEnum.MINIAPP)]
    public class ImMessage
    {
        /// <summary>
        /// 主键id
        /// </summary>
        [SqlField(IsPrimaryKey = true, IsAutoId = true)]
        public int Id { get; set; } = 0;

        [SqlField]
        public string appId { get; set; } = string.Empty;

        /// <summary>
        /// 
        /// </summary>
        [SqlField]
        public int aId { get; set; } = 0;

        /// <summary>
        /// 门店id
        /// </summary>
        [SqlField]
        public int storeId { get; set; } = 0;
        /// <summary>
        /// 房间id
        /// </summary>
        [SqlField]
        public int roomId { get; set; } = 0;
        /// <summary>
        /// 发送者id
        /// </summary>
        [SqlField]
        public int fuserId { get; set; } = 0;
        /// <summary>
        /// 接收者id
        /// </summary>
        [SqlField]
        public int tuserId { get; set; } = 0;
        /// <summary>
        /// 状态 0:正常
        /// </summary>
        [SqlField]
        public int state { get; set; } = 0;
        /// <summary>
        /// 消息内容
        /// </summary>
        [SqlField]
        public string msg { get; set; } = string.Empty;
        /// <summary>
        /// 发送时间
        /// </summary>
        [SqlField]
        public string sendDate { get; set; } = string.Empty;
        /// <summary>
        /// 添加时间
        /// </summary>
        [SqlField]
        public string createDate { get; set; } = string.Empty;
        /// <summary>
        /// 修改时间
        /// </summary>
        [SqlField]
        public string updateDate { get; set; } = string.Empty;

        /// <summary>
        /// 用户类型
        /// </summary>
        [SqlField]
        public int tuserType { get; set; } = 0;
        /// <summary>
        /// 阅读状态 0未读 1已读
        /// </summary>
        [SqlField]
        public int isRead { get; set; } = 0;
        /// <summary>
        /// uuid
        /// </summary>
        [SqlField]
        public string ids { get; set; } = string.Empty;
        /// <summary>
        /// 消息类型 0文字 1图片
        /// </summary>
        [SqlField]
        public int msgType { get; set; } = 0;

        /// <summary>
        /// 名片姓名
        /// </summary>
        public string CardName { get; set; }
        /// <summary>
        /// 名片头像
        /// </summary>
        public string CardImgUrl { get; set; }
        public string TUserImgUrl { get; set; }
        public string FUserImgUrl { get; set; }
        /// <summary>
        /// 未读私信信息数量
        /// </summary>
        public int NoReadMessageCount { get; set; }
        /// <summary>
        /// 最后一条私信信息
        /// </summary>
        public string LastMsg { get; set; }
        /// <summary>
        /// 客服姓名
        /// </summary>
        public string EmployeeName { get; set; }
        /// <summary>
        /// 客户备注
        /// </summary>
        public string Desc { get; set; }
        public string StoreName { get; set; }
        public string StoreLogImgUrl { get; set; }
    }
}
